src/devices/src/virtio/net.rs

```diff
@@ -229,9 +229,9 @@ impl NetEpollHandler {
                     let write_result = self.mem.write_slice(source_slice, desc.addr);
 
                     match write_result {
-                        Ok(sz) => {
+                        Ok(()) => {
                             METRICS.net.rx_count.inc();
-                            write_count += sz;
```

Comment:
With this change the behavior of this function changes. Before we were not returning an error if only a part of the slice was successfully written. Probably here we want to check if the Error is PartialBuffer and get the number of actual bytes written; then we need to increment the write count.

Reply:
Done.